exonum/src/node/consensus.rs

```diff
@@ -245,33 +223,41 @@ impl NodeHandler {
         }
 
         if self.state.block(&block_hash).is_none() {
-            // Verify transactions
-            let tx_hashes = if let Some(res) = self.validate_block_transactions(msg) {
-                self.blockchain
-                    .merge(res.1)
-                    .expect("Unable to save transaction to persistent pool.");
-                res.0
-            } else {
+            if !msg.verify_tx_hash() {
+                error!("Received block has wrong root hash, msg={:?}", msg);
```

Comment:
"root hash" is strange message, can we rewrite it as "received block with invalid tx_hashes" or something more DevOps friendly?